src/components/FormInputs/EditorToolbar.tsx

```diff
@@ -59,16 +59,27 @@ export default function EditorToolbar({ editor, isFocussed }: Props) {
       textInputProps: { placeholder: 'https://example.com/image.png' },
       okText: 'Save',
       closeOnOkClick: false,
-      okCallback: (url) => {
-        if (url && isValidImageUrl(url)) {
-          // update link
-          editor.chain().focus().setImage({ src: url }).run();
-          showPrompt(null);
-        } else {
-          showAlert({
-            message: 'Invalid image URL! An image url must end with a valid image file extension',
-            type: 'warning',
-          });
+      okCallback: async (url) => {
+        if (url) {
+          try {
+            const response = await fetch(url, { method: 'HEAD' });
+
+            if (response.ok && response.headers.get('content-type')?.includes('image')) {
+              // update link
+              editor.chain().focus().setImage({ src: url }).run();
+              showPrompt(null);
+            } else {
+              showAlert({
+                message: 'Invalid image URL',
+                type: 'warning',
+              });
+            }
+          } catch (err) {
+            showAlert({
+              message: 'Not able to load the image',
```

Comment:
What's the different between this and the above error message? Should both messages be the same?

Reply:
it will go in catch if url is correct but content is not loadable, above error message will show if url is not correct

Reply:
Ohh. And what happens if network error?

Reply:
coming in catch if if network error

Reply:
OK